internal/exports/disk_writer.go

```diff
@@ -61,9 +57,51 @@ func DiskCallbackFactory(
 	var statOut csvOutput = nil
 	var pmOut csvOutput = nil
 
+	if writeImages || writeTimeseries {
+		// Create Directory and File
+		err := os.MkdirAll(output, os.ModePerm)
+		if err != nil {
+			log.Printf("Could not create output directory '%v'", output)
+		}
+	}
+
 	callback := func(pkg common.ExportablePackage) {
+		err = pkg.ParsingError()
+		if err != nil {
+			log.Println(err)
+		}
 		if writeImages {
-			// TODO: Add image writing here
+			switch pkg.AEZData().(type) {
+			case aez.CCDImage:
+				ccdImage, ok := pkg.AEZData().(aez.CCDImage)
+				if !ok {
+					log.Print("Could not understand packet as CCDImage, this should be impossible.")
+					break
+				}
+				imgFileName := getGrayscaleImageName(output, ccdImage.PackData)
+
+				imgData := getImageData(
+					pkg.RemainingBuffer(),
+					ccdImage.PackData,
+					imgFileName,
+				)
+				_, shift, _ := ccdImage.PackData.WDW.InputDataWindow()
+				img := getGrayscaleImage(
+					imgData,
+					int(ccdImage.PackData.NCOL+aez.NCOLStartOffset),
+					int(ccdImage.PackData.NROW),
+					shift,
+					imgFileName,
+				)
+
+				imgFile, err := os.Create(imgFileName)
+				if err != nil {
+					log.Printf("failed creating %s: %s", imgFileName, err)
+					panic(err.Error())
+				}
+				defer imgFile.Close()

```

Comment:
```suggestion
				imgFile, err := os.Create(imgFileName)
				defer imgFile.Close()
				if err != nil {
					log.Printf("failed creating %s: %s", imgFileName, err)
					panic(err.Error())
				}

```
Or is that bad, i.e. will `Close` in itself likely cause an error?

Reply:
no that should be fine